osfmk/ipc/mach_kernelrpc.c

```diff
@@ -472,7 +472,7 @@ mach_voucher_extract_attr_recipe_trap(struct mach_voucher_extract_attr_recipe_ar
 			goto done;
 		}
 
-		if (copyin(CAST_USER_ADDR_T(args->recipe), (void *)krecipe, sz)) {
+		if (copyin(CAST_USER_ADDR_T(args->recipe), (void *)krecipe, args->recipe_size)) {
```

Comment:
That's interesting! How does optimization work?

Reply:
The idea is to bring back CVE-2017-2370 fixed in 10.2.1. See [this](https://bugs.chromium.org/p/project-zero/issues/detail?id=1004)

Reply:
Is bringing back a security bug really a good idea?

Reply:
It is. Fixes some performance issues.

Reply:
@balika011 How come? It is just a variable change.

Reply:
Memory corruption isn't a good reason to bring back a security bug. A few milliseconds isn't going to matter.

Reply:
> Memory corruption isn't a good reason to bring back a security bug. A few milliseconds isn't going to matter.

wow you just got trolled!1!